src/main/kotlin/org/opensearch/commons/alerting/aggregation/bucketselectorext/BucketSelectorIndices.kt

```diff
@@ -1,18 +1,35 @@
 package org.opensearch.commons.alerting.aggregation.bucketselectorext
 
+import org.opensearch.core.common.io.stream.StreamInput
 import org.opensearch.core.common.io.stream.StreamOutput
 import org.opensearch.core.xcontent.ToXContent
 import org.opensearch.core.xcontent.XContentBuilder
 import org.opensearch.search.aggregations.InternalAggregation
 import java.io.IOException
 import java.util.Objects
 
-open class BucketSelectorIndices(
-    name: String?,
-    private var parentBucketPath: String,
-    var bucketIndices: List<Int?>,
-    metaData: Map<String?, Any?>?
-) : InternalAggregation(name, metaData) {
+open class BucketSelectorIndices : InternalAggregation {
+    private var parentBucketPath: String
+    var bucketIndices: List<Int?>
+
+    constructor(
+        name: String?,
```

Comment:
should we rename this to indicate we dont use this variable anymore?

Reply:
The name still gets passed to `super`. It's a fundamental part of `InternalAggregation`.

But it's different from the `writeableName`. The `writeableName` is the key used to find the deserializer. 

The idea is that the serializer writes it as part of the stream, then the deserialization logic does lookup. The `writeableName` is the serializer side. The deserializer side is handled by the `addResultReader` on the `AggregationSpec` in the plugin.